lib/internal/test_runner/runner.js

```diff
@@ -153,92 +153,58 @@ function getRunArgs({ path, inspectPort, testNamePatterns }) {
   return argv;
 }
 
+const serializer = new v8.DefaultSerializer();
+serializer.writeHeader();
+const v8Header = serializer.releaseBuffer();
+const v8HeaderAndSize = 4 + v8Header.length;
+
 class FileTest extends Test {
   #buffer = [];
+  #messageBuffer = [];
+  #messageBufferSize = 0;
   #reportedChildren = 0;
   failedSubtests = false;
   #skipReporting() {
     return this.#reportedChildren > 0 && (!this.error || this.error.failureType === kSubtestsFailed);
   }
-  #checkNestedComment({ comment }) {
+  #checkNestedComment(comment) {
     const firstSpaceIndex = StringPrototypeIndexOf(comment, ' ');
     if (firstSpaceIndex === -1) return false;
     const secondSpaceIndex = StringPrototypeIndexOf(comment, ' ', firstSpaceIndex + 1);
     return secondSpaceIndex === -1 &&
           ArrayPrototypeIncludes(kDiagnosticsFilterArgs, StringPrototypeSlice(comment, 0, firstSpaceIndex));
   }
-  #handleReportItem({ kind, node, comments, nesting = 0 }) {
-    if (comments) {
-      ArrayPrototypeForEach(comments, (comment) => this.reporter.diagnostic(nesting, this.name, comment));
+  #handleReportItem(item) {
+    if (item.type === 'test:plan' && item.data.nesting === 0 && this.#skipReporting()) {
```

Comment:
It might be slightly faster (assuming these are not the common case) to structure this conditional and the following one like:

```js
if (item.data.nesting === 0) {
  if (item.type === 'test:plan' && ...) {
    return;
  }

  if (item.type === 'test:diagnostic' && ...) {
    return;
  }
}
```